design/design-doc-rbac-manager.md

```diff
@@ -0,0 +1,843 @@
+  controllers run with the ability to escalate privileges.
+
+### Managed RBAC ClusterRoles
+
+The RBAC manager will be responsible for the management of several cluster
+roles, using [`ClusterRole` aggregation][Aggregated ClusterRoles] to form a set
+of cluster roles that may change over time as providers and composite resources
+are added to and removed from the API server. The RBAC manager would manage the
+following cluster roles. Note that these roles are aggregated - they consist of
+several more granular roles that are not covered here. **The following examples
+are 'flattened' aggregations represented as a single fixed `ClusterRole` to
+demonstrate specifically what permissions each role will have.**
+
+The Crossplane `ClusterRole` is bound to the service account the Crossplane pod
+runs as. It grants the following rules:
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: crossplane
+rules:
+# Crossplane creates events to report its status.
+- apiGroups: [""]
+  resources: [events]
+  verbs: [create]
+# Crossplane manages the secrets of its composite resources. It never explicitly
+# deletes secrets, but instead relies on garbage collection.
+- apiGroups: [""]
+  resources: [secrets]
+  verbs: [get, create, update]
+# Crossplane creates CRDs in response to the creation of XRDs.
+- apiGroups: [apiextensions.k8s.io]
+  resources: [customresourcedefinitions]
+  verbs: [get, create, update, delete]
+# Crossplane has full access to the types it defines.
+- apiGroups:
+  - apiextensions.crossplane.io
+  - pkg.crossplane.io
+  resources: ["*"]
+  verbs: ["*"]
+# Crossplane has full access to the composite resources and claims it defines.
+- apiGroups: [xr.example.org]
+  resources:
+  - examplecomposites
+  - examplecomposites/status
+  - exampleclaims
+  - exampleclaims/status
+  verbs: ["*"]
+# Crossplane has access to the provider resource that it may need to compose.
+- apiGroups: [provider.example.org]
+  resources:
+  - examplemanageds
+  - exampleproviderconfigs
+  verbs: ["*"]
+---
+# Crossplane uses a fixed (at runtime) namespace to unpack packages and run
+# providers. This role allows it to do so. Note that it is included for
+# illustrative purposes; it is static and is thus managed by the Crossplane Helm
+# chart rather than the RBAC manager.
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  namespace: crossplane-system
+  name: crossplane
+rules:
+# Crossplane uses jobs to unpack packages.
+- apiGroups: [batch, extensions]
+  resources: [jobs]
+  verbs: [get, create, update, delete]
+# Crossplane reads pod logs to unpack packages.
+- apiGroups: [""]
+  resources: [pods]
+  verbs: [get, list]
+- apiGroups: [""]
+  resources: [pods/log]
+  verbs: [get]
+# Crossplane uses deployments to run providers.
+- apiGroups: [extensions, apps]
+  resources: [deployments]
+  verbs: [get, create, update, delete]
```

Comment:
I have ended up needing some additional permissions here (such as `watch`), but these could change depending on final details of package manager implementation, so I would defer to me updating this after that is merged.

Reply:
I'll add list and watch for now. I agree we can update this later to more accurately reflect reality as we understand it better.